feature-toggles/test/specs/auth/authServiceSpec.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+(describe('Test AuthService', function() {
+    let authService, userService, userFactory;
+
+    let userTest = {
+        name : 'User',
+        accessToken: 'gfdfggfdjdsfkbcbmnweuiyeuiwyhdjskalhdjkhjk',
+        emailVerified: true
+    };
+
+    let firebaseUser = {
+        accessToken: 'ruioewyuirywieuryiuweyr876324875632487yiue',
+        getIdToken: async () => firebaseUser.accessToken
+    }
+
+    beforeEach(module('app'));
+
+    beforeEach(inject(function(AuthService, UserService, UserFactory) {
+        authService = AuthService;
+        userService = UserService;
+        userFactory = UserFactory;
+        
+        firebase.auth = () => {
+            return {
+                onAuthStateChanged: (callback) => callback(firebaseUser),
+                signOut: function signOut() {}
+            };
+        }
+
+        firebase.auth.GoogleAuthProvider = function GoogleAuthProvider() {};
+        authService.useOriginalGetUserToken();
+    }));
+
+    describe('AuthService  setupUser', function() {
+
+        it('should call authService.setupUser()', function() {
+            spyOn(userService, 'load').and.callThrough();
+
+            authService.setupUser(userTest.accessToken, userTest.emailVerified);
+            var user = authService.getCurrentUser();
+            var new_user = new userFactory.user(userTest);
+
+            expect(userService.load).toHaveBeenCalled();
+            expect(user).toEqual(new_user);
+        });
+    });
+
+    describe('AuthService user informations', function() {
+        beforeEach(function() {
+            authService.setupUser(userTest.accessToken);
```

Comment:
why in this beforeEach function you call setupUser with no 'emailVerified' parameter, and then in the most 'it' tests you call setupuser again?